PHPFUI ORM

phpfui opened the pull request:
I added my ORM, phpfui/orm.  I was quite impressed with the performance.  I believe the logic is correct.  I patterned off the Eloquent source.

Also updated to more modern PHP.  Eloquent is now the latest.  I removed the vendor files, which should never be checked in.

The other ORMs don't seem to work.  I would update them to the latest, but not sure how they all work.  Probably not that difficult, just no idea of how to fix them.

phpfui committed:
dirname(__FILE__) to __DIR__

phpfui committed:
Delete vendor and lock files

phpfui committed:
Upgrade Eloquent to v10

phpfui committed:
Ignore vendor and composer.lock files

phpfui committed:
Add PHPFUI_ORM

phpfui committed:
Remove old sqlite memory db

phpfui committed:
Updating other ORM benchmarks

adrianmiu commented:
@phpfui Sorry for the delayed response. I've checked your code and I've found the following:
1. On the first run I got zero memory usage which made me investigate the tests
2. The `insert()` method doesn't seem to actually insert the record into the DB. I've added things like `$product->insert()` and `$category->insert()` and the time went from 22ms to 36ms and memory usages at around 8Mb
3. The eager loading test is empty. It is supposed to time the retrieval of products with related objects.

phpfui commented:
I'll revisit.  I am not sure I fully implemented things correctly for the benchmarks.  I may have questions and will let you know.

Thanks for the followup.

phpfui commented:
Yes, the insert was missing.  Fixed.

I had also update the eager loading test before, but forgot to push it.  Should be OK now.

The reason the memory usage is zero or near zero is because PHPFUI/ORM is not a repository model with collections of records.  It is an Active Record / Active Table model with Cursors to minimize memory usage, which is the bane of ORMs in general and why I decided to write an ORM. It will map very closely to raw SQL because it generates raw SQL from an interface that models SQL itself.  This is not an abstract ORM that can use any database structure, it can only be used on an SQL schema, like 99% of all apps use, so there was no need to slow it down making it generic to any DB schema.

I also noticed it seems that all tests are using SQLite memory.  I think you intended to test both disk and memory.  Not sure if I broke that (don't think so), but worth looking into.

I also updated some of the dependencies, but several ORMs seem to be broken when I tried them.  Missing classes or PHP errors.  I commented them out of the main test file.  I would fix these, but no idea how and it seems many are no longer supported anyway.

Let me know of any questions / issues you have.

Thanks for posting and maintaining an ORM benchmark.

phpfui committed:
Upgrade to ORM V2

phpfui commented:
I upgraded my ORM to V2. It needed some new class generations. I believe it correctly implements the benchmark.

I also noticed that a bunch of the other ORMs tested seem to be broken and / or on older versions.  Let me know if you want some help updating them.

adrianmiu commented:
@phpfui I would greatly appreciate if you would update the other ORMs

phpfui commented:
OK. Will do.

Bruce

On Fri, Oct 17, 2025, 11:54 AM Adrian Miu ***@***.***> wrote:

> *adrianmiu* left a comment (adrianmiu/forked-php-orm-benchmark#6)
> <https://github.com/adrianmiu/forked-php-orm-benchmark/pull/6#issuecomment-3416139874>
>
> @phpfui <https://github.com/phpfui> I would greatly appreciate if you
> would update the other ORMs
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adrianmiu/forked-php-orm-benchmark/pull/6#issuecomment-3416139874>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYW6S3PV3JQEKWTOCN7YM33YEGLTAVCNFSM6AAAAACJPU6XO6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTIMJWGEZTSOBXGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
